### PR TITLE
[Console] reopen inputStream after a ctrl+D/ctrl+Z

### PR DESCRIPTION
**Pull request by epitre**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | I did not open a ticket
| License       | MIT
| Doc PR        | N/A

I encountered a problem with the new multiline question.

it appears when you ask a question AFTER a multiline question

It seems STDIN needs to be reopen (<- does it make sense ?).

I have no idea if this is the best way to fix it, but it does fix it ^^

### CODE REVIEW

File: src/Symfony/Component/Console/Helper/QuestionHelper.php [hunk 1]
```diff
@@ -521,6 +521,7 @@ private function readInput($inputStream, Question $question)
         while (false !== ($char = fgetc($inputStream))) {
             $ret .= $char;
         }
+        $this->inputStream = fopen('php://stdin', 'r');
```
Comment (stof) [inline src/Symfony/Component/Console/Helper/QuestionHelper.php:524]: what if the input stream being used is *not* `php://stdin` ? That's a supported use case (see `isInteractiveInput` supporting that case explicitly for instance).

Also, this method looks like the wrong place to me, as it is not designed to use `$this->inputStream` directly.

**Comment by ramsey** [comment]
@epitre Thanks for looking into this. I'm going to do some testing myself, since I think there's a more appropriate way to "reset" the input stream, since as @stof points out, the input stream being used might not be `stdin`.

File: src/Symfony/Component/Console/Helper/QuestionHelper.php (hunk 1 above)
Comment (epitre) [reply src/Symfony/Component/Console/Helper/QuestionHelper.php:524]: That's a very valid point.
I have no idea how to deal with it ^^

Comment (ramsey) [reply src/Symfony/Component/Console/Helper/QuestionHelper.php:524]: I think I have a solution. I'm testing it on Windows to make sure it works there, as well.

**Comment by ramsey** [comment]
It took a while since what I originally had worked on *nix but not on Windows, but I have a solution over here: https://github.com/symfony/symfony/pull/38351

**Comment by fabpot** [comment]
Closing in favor of #38351

**Comment by ramsey** [comment]
@epitre, this fix for this was merged into master just now (see https://github.com/symfony/symfony/pull/38351). Can you pull that down and test to make sure this resolves the issue? Thanks!

**Comment by epitre** [comment]
@ramsey It works perfectly !
Thank you very much :)

**Pull request closed without merging**